File: runtime/go/units/rt/i32_div_s.go
// requires: rt/trap
func (rt) i32_div_s(a, b uint32) uint32 {
    if b == 0 {
        Rt.trap("integer divide by zero")
    }
    if a == 0x80000000 && b == 0xffffffff {
        Rt.trap("integer overflow")
    }
    return uint32(int32(a) / int32(b))
}

File: crates/dewasm-backend-python/units/rt/i32_div_s.py
# requires: rt/s32, rt/trap
@staticmethod
def i32_div_s(a, b):
    sa = Rt.s32(a)
    sb = Rt.s32(b)
    if sb == 0:
        Rt.trap("integer divide by zero")
    q = abs(sa) // abs(sb)
    if (sa < 0) ^ (sb < 0):
        q = -q
    if q > 0x7FFFFFFF:
        Rt.trap("integer overflow")
    return q & Rt.M32
